README.markdown

```diff
@@ -1,26 +1,59 @@
 
-* sign up with cloudinary and get the cloudinary.yml file. Replace it with cloudinary.yml.example.
-* Use `rake secret` to generate the secret key, copy the generated key and paste it for `devise_key`. Do the same for `secret_key_base`
-* If you would like to use fb login too then sign up with fb developers site, get your app key and app secret and paste it in `fb_app` and `fb_secret`.
-* Rename the database.rb.example file present in the config/initializers directory to database.rb . 
-* Rename the production.rb.example file in config/environments directory to production.rb
-* For database settings you will have to fill in the username and password of your own local env pg.
+* For Debian/Ubuntu: `sudo apt-get install postgresql postgresql-contrib libpq-dev`
+* Fedora/Red Hat/CentOS: `sudo yum install postgresql-server postgresql-contrib postgresql-devel`
+* Mac OS X (using brew): `brew install postgres`
+
+## Image uploads
+We are using cloudinary as a cloud storage for images. So signup for a cloudinary account so that you can get the cloud name, api key and secret. We are using `carrierwave` to manage uploads to cloudinary.
+
+# DEPLOYMENT
+
+* We have all our secrets in secret.yml.example. You only need to rename it to secret.yml and fill in the secrets
+* Use `rake secret` to generate the secret key, copy the generated key and paste it for `devise_key`. Do the same for `secret_key_base`. Enter the _cloud_name_, _api_key_ and _api_secret_ that you got from cloudinary in the `cloud_name`, `cloud_api_key` and `cloud_api_secret` fields
+* If you would like to use fb login too then sign up with fb developers site, get your app key and app secret and paste it in `fb_app` and `fb_secret`
+* Rename the database.rb.example file present in the config/initializers directory to database.rb
+* For database settings you will have to fill in the username and password of your own local pg env.
 
 # Running your local instance
 
 * `cd` to app directory
 * Run `bundle install` to install all the necessary gems
-* Create databse with `rake db:create`
-* Migrate the database with `rake db:migrate`
+* For first time installation run `rake db:setup` to setup the database
 * Start the server with `rails s`
-* Go to `localhost:3000` in your broswer.
+* Go to `localhost:3000` in your broswer
+
+## Using database migrations
+What `rake db:setup` does is that it creates the development and test database using `rake db:create`, populates the database using the schema.rb file and then seeds the data using `rake db:seed`. When a new migration is introduced you don't need to run `rake db:setup` again, instead run `rake db:migrate` to run the migrations and update the database schema.
+
+# Testing
+Run tests using the command `rspec` or `bundle exec rspec`. We are using Rspec as your Rails test suite. Writing good tests ensure that all new features introduced don't break previous ones. So consider writing tests  whenever you work on new features.
```

Comment:
Uh it is TDD framework. So how does it sound if I put its as `We are using Rspec as our testing framework` ?